### PR TITLE
Fix the issues allowing the current site situation

### PR DESCRIPTION
**Pull request by woaisthisgit**
This is the first of many fixes, please do not merge

### CODE REVIEW

**Comment by woaisthisgit** [comment]
This *should* be enough to prevent what's happening right now but I plan to also fix the root of the error.

**Comment by woaisthisgit** [comment]
Haven't tested any of this but should all be fine

File: src/gameplay/room.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function Room(
     newRoomPassword_ = undefined;
   }
 
-  if (maxNumPlayers_ === '' || maxNumPlayers_ < 5 || maxNumPlayers_ > 10) {
+  if (!maxNumPlayers_.isInteger() || maxNumPlayers_ < 5 || maxNumPlayers_ > 10) {
```
Comment (vck3000) [inline src/gameplay/room.js:19]: I'm not sure if `isInteger()` is a valid method on a string object. A better way might be to use `parseInt` and check for `NaN`. I'll do that now.

File: src/gameplay/room.js [hunk 2]
```diff
@@ -373,6 +373,7 @@ Room.prototype.getSocketsOfSpectators = function () {
 Room.prototype.updateMaxNumPlayers = function (socket, number) {
   if (
     socket.request.user.username === this.host &&
+    number.IsInteger() &&
```
Comment (vck3000) [inline src/gameplay/room.js:376]: `isInteger`, not `IsInteger` :). But ditto here as well.

File: src/sockets/sockets.ts [hunk 3]
```diff
@@ -2782,8 +2782,8 @@ function newRoom(dataObj) {
 function joinRoom(roomId, inputPassword) {
   // console.log("inputpassword: " + inputPassword);
 
-  // if the room exists
-  if (rooms[roomId]) {
+  // if the room exists and the player is not currently in a room
+  if (rooms[roomId] && this.request.user.inRoomId === undefined) {
```
Comment (vck3000) [inline src/sockets/sockets.ts:2786]: Would prefer `!this.request.user.inRoomId` :). 
Helps us catch the case where it's 0, or null as well.

**Pull request merged**